### PR TITLE
whisper: Exclude Whisper model files from backups

### PR DESCRIPTION
**Pull request by tronikos**
Exclude large Whisper model files from backups. I'm using the faster-whisper-tiny-int8 which is ~42MB or ~34MB compressed.

This is what I see inside the add on:

```
root@core-whisper:/# ls -Rla data/models*
data/models--rhasspy--faster-whisper-tiny-int8:
total 20
drwxr-xr-x 5 root root 4096 Apr  7 05:37 .
drwxr-xr-x 4 root root 4096 May 24 13:06 ..
drwxr-xr-x 2 root root 4096 Apr  7 05:37 blobs
drwxr-xr-x 2 root root 4096 Apr  7 05:37 refs
drwxr-xr-x 3 root root 4096 Apr  7 05:37 snapshots

data/models--rhasspy--faster-whisper-tiny-int8/blobs:
total 41600
drwxr-xr-x 2 root root     4096 Apr  7 05:37 .
drwxr-xr-x 5 root root     4096 Apr  7 05:37 ..
-rw-r--r-- 1 root root 42120343 Apr  7 05:37 6fdeb89775be8b3455197e4685fec5d59ce47874de4fb8fb7724a43ea205a7b0
-rw-r--r-- 1 root root   459861 Apr  7 05:37 c9074644d9d1205686f16d411564729461324b75
-rw-r--r-- 1 root root     2020 Apr  7 05:37 e86f0ed3287af742920a8ef37577c347a1a8153b

data/models--rhasspy--faster-whisper-tiny-int8/refs:
total 12
drwxr-xr-x 2 root root 4096 Apr  7 05:37 .
drwxr-xr-x 5 root root 4096 Apr  7 05:37 ..
-rw-r--r-- 1 root root   40 May 24 13:07 main

data/models--rhasspy--faster-whisper-tiny-int8/snapshots:
total 12
drwxr-xr-x 3 root root 4096 Apr  7 05:37 .
drwxr-xr-x 5 root root 4096 Apr  7 05:37 ..
drwxr-xr-x 2 root root 4096 Apr  7 05:37 5b6382e0f4ac867ce9ff24aaa249400a7c6c73d9

data/models--rhasspy--faster-whisper-tiny-int8/snapshots/5b6382e0f4ac867ce9ff24aaa249400a7c6c73d9:
total 12
drwxr-xr-x 2 root root 4096 Apr  7 05:37 .
drwxr-xr-x 3 root root 4096 Apr  7 05:37 ..
lrwxrwxrwx 1 root root   52 Apr  7 05:37 config.json -> ../../blobs/e86f0ed3287af742920a8ef37577c347a1a8153b
lrwxrwxrwx 1 root root   76 Apr  7 05:37 model.bin -> ../../blobs/6fdeb89775be8b3455197e4685fec5d59ce47874de4fb8fb7724a43ea205a7b0
lrwxrwxrwx 1 root root   52 Apr  7 05:37 vocabulary.txt -> ../../blobs/c9074644d9d1205686f16d411564729461324b75
```

And this is what I see in the backup:

image

image

## Summary by CodeRabbit

- **New Features**
  - Improved the backup process by excluding certain files ("data/models*") from backups.
- **Updates**
  - Updated configuration to version 2.1.1.

### CODE REVIEW

**Review by frenck — CHANGES_REQUESTED** [review]
Looks good to me! 

Can you also bump the version and update the changelog?

../Frenck

File: whisper/config.yaml [hunk 1]
```diff
@@ -12,6 +12,7 @@ discovery:
   - wyoming
 backup_exclude:
   - "*.bin"
```
Comment (tronikos) [inline whisper/config.yaml:14]: @synesthesiam can this be removed? You added this in #3116 but it doesn't seem to do anything, at least for the model I'm using.

Comment (tronikos) [reply whisper/config.yaml]: I tried many different models and I didn't see any bin files so I removed this line. The new pattern seems to capture all the models. I forgot to delete all the unused the models after testing and my last backup is 4 GB.

Comment (synesthesiam) [reply whisper/config.yaml]: I changed the source of the models in a previous version, but forgot to update the backup exclusion. Thank you for this.

**Review by synesthesiam — APPROVED** [review]

**Comment by synesthesiam** [comment]
@frenck This looks good to me :+1:  I can't merge things in this repo.

**Review by agners — APPROVED** [review]

**Comment by rwjack** [comment]
Layman's perspective, maybe it needs to be:
```
- "data/models/*"
```

**Comment by tronikos** [comment]
Not sure why this didn't work. I tested it with the following code which was copied from the supervisor codebase:
image
I'll change it to `models*` which I've verified it works on a local addon which is a copy of the whisper configuration with only this change.

**Pull request merged**